File: src/main/java/com/company/erp/features/auth/dto/AuthResponse.java
package com.company.erp.features.auth.dto;

import com.company.erp.features.users.dto.UserDto;

import java.time.Instant;

public record AuthResponse(
        String accessToken,
        Instant accessTokenExpiresAt,
        String refreshToken,
        Instant refreshTokenExpiresAt,
        UserDto user
) {
}

File: app/Features/Auth/Dto/AuthResponse.php
<?php

namespace App\Features\Auth\Dto;

/** Port of the Spring AuthResponse record. */
final class AuthResponse
{
    /**
     * @param array<string, mixed> $user  UserDto array
     * @return array<string, mixed>
     */
    public static function build(
        string $accessToken,
        int $accessTokenExpiresAt,
        string $refreshToken,
        int $refreshTokenExpiresAt,
        array $user
    ): array {
        return [
            'accessToken' => $accessToken,
            'accessTokenExpiresAt' => self::iso($accessTokenExpiresAt),
            'refreshToken' => $refreshToken,
            'refreshTokenExpiresAt' => self::iso($refreshTokenExpiresAt),
            'user' => $user,
        ];
    }

    private static function iso(int $epochSeconds): string
    {
        return gmdate('Y-m-d\TH:i:s\Z', $epochSeconds);
    }
}
